app/controllers/flags_controller.rb

```diff
@@ -17,11 +17,24 @@ def destroy
     flag = answer.flags.find(params[:id])
     if current_user.has_flagged? answer or current_user.admin?
       flag.destroy
-      redirect_to request.referrer
+      refresh_term_page(params[:term_id])
```

Comment:
if we keep 'redirect_to request.referrer' then if we're on the flags page or the term page then we just just go back to where we are which I think is better?